File: src/test/java/easy/StringIsSubsequenceTest.java
package easy;

import static org.junit.jupiter.api.Assertions.*;

import java.util.stream.Stream;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

public class StringIsSubsequenceTest {
  StringIsSubsequence instance = new StringIsSubsequence();

  @ParameterizedTest
  @MethodSource("dataStringIsSubsequence")
  public void testStringIsSubsequence(String s, String t, boolean expected) {

    boolean result = instance.solve(s, t);

    assertEquals(expected, result);
  }

  public static Stream<Arguments> dataStringIsSubsequence() {
    return Stream.of(
        Arguments.of("abc", "ahbgdc", true),
        Arguments.of("", "", true),
        Arguments.of("", "anything", true),
        Arguments.of("a", "a", true),
        Arguments.of("a", "b", false),
        Arguments.of("z", "abc", false),
        Arguments.of("acb", "abc", false),
        Arguments.of("xyz", "abcxyz", true),
        Arguments.of("aaa", "aaaaaa", true),
        Arguments.of("aaa", "aa", false),
        Arguments.of("abc", "abc", true),
        Arguments.of("abc", "a" + "x".repeat(9999) + "b" + "y".repeat(9999) + "c", true),
        Arguments.of("ace", "abcde", true),
        Arguments.of("aec", "abcde", false),
        Arguments.of("axc", "ahbgdc", false));
  }
}

File: easy/string_is_subsequence_test.go
package easy

import (
	"strings"
	"testing"
)

func TestStringIsSubsequence(t *testing.T) {
	cases := []struct {
		s        string
		t        string
		expected bool
	}{

		{"abc", "ahbgdc", true},
		{"", "", true},
		{"", "anything", true},
		{"a", "a", true},
		{"a", "b", false},
		{"z", "abc", false},
		{"acb", "abc", false},
		{"xyz", "abcxyz", true},
		{"aaa", "aaaaaa", true},
		{"aaa", "aa", false},
		{"abc", "abc", true},
		{"abc", "a" + strings.Repeat("x", 9999) + "b" + strings.Repeat("y", 9999) + "c", true},
		{"ace", "abcde", true},
		{"aec", "abcde", false},
		{"axc", "ahbgdc", false},
	}
	for _, c := range cases {
		t.Run("", func(t *testing.T) {
			result := StringIsSubsequence(c.s, c.t)
			if result != c.expected {
				t.Errorf("StringIsSubsequence(%v, %v) == %v, expected %v", c.s, c.t, result, c.expected)
			}

		})

	}
}
